packages/cozy-client/src/store/documents.js

```diff
@@ -107,3 +92,61 @@ export const getDocumentFromSlice = (state = {}, doctype, id) => {
   }
   return state[doctype][id]
 }
+
+const extractAndMergeDocument = (data, updatedStateWithIncluded) => {
+  const doctype = data[0]._type
+
+  if (!doctype) {
+    throw new Error('Document without _type', data[0])
+  }
+  const ordonnedData = keyBy(data, properId)
+
+  let mergedData = {}
+  if (updatedStateWithIncluded && updatedStateWithIncluded[doctype]) {
+    Object.values(updatedStateWithIncluded[doctype]).map(dataState => {
+      if (!mergedData[doctype]) mergedData[doctype] = {}
+      if (ordonnedData[dataState._id]) {
+        mergedData[doctype] = {
+          ...mergedData[doctype],
```

Comment:
Is there a point to spreading the object every time? ` mergedData[doctype][dataState._id] = …` would work as well, right?